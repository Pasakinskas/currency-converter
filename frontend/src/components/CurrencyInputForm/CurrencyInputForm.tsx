import React from 'react';
import {convertCurrency} from "../../services/CurrencyConversionService";

export function CurrencyInputForm() {
  const [initialCurrency, setInitialCurrency] = React.useState();
  const [targetCurrency, setTargetCurrency] = React.useState();
  const [initialAmount, setInitialAmount] = React.useState();
  const [convertedAmount, setConvertedAmount] = React.useState();

  const convert = () => {
    console.log(initialAmount, initialCurrency, targetCurrency);
  }
  return (
    <div className="pure-form pure-form-aligned">
      <div className="pure-control-group">
        <label htmlFor="aligned-name">Initial Amount</label>
        <input
          className="pure-u-1-5"
          type="number"
          step="0.01"
          placeholder="Initial amount"
          required
          onChange={e => setInitialAmount(e.target.value)}
        />
      </div>
      <div className="pure-control-group">
        <label htmlFor="Initial Currency">Initial Currency</label>
        <select
          id="Initial Currency"
          className="pure-u-1-5"
          onChange={e => setInitialCurrency(e.target.value)}
        >
          <option>EUR</option>
          <option>GBP</option>
          <option>USD</option>
        </select>
      </div>
      <div className="pure-control-group">
        <label htmlFor="Target Currency">Target Currency</label>
        <select
          id="Target Currency"
          className="pure-u-1-5"
          onChange={e => setTargetCurrency(e.target.value)}
        >
          <option>EUR</option>
          <option>GBP</option>
          <option>USD</option>
        </select>
      </div>
      <div className="pure-control-group">
        <label htmlFor="amount-after-conversion">Amount after Conversion</label>
        <input
          className="pure-u-1-5"
          id="amount-after-conversion"
          type="text"
          value={convertedAmount}
          readOnly={true}
        />
      </div>
      <div className="pure-controls">
        <button onClick={convert} className="pure-button pure-button-primary">Calculate</button>
      </div>
    </div>
  );
}