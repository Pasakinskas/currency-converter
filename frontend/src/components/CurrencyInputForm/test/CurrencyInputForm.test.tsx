import React from "react";
import TestRenderer from 'react-test-renderer';

import { CurrencyInputForm } from "../CurrencyInputForm";

describe("currencyInputForm", () => {
  test("renders correctly", () => {
    const inputForm = <CurrencyInputForm />
    const renderedInputForm = TestRenderer
      .create(inputForm)
      .toJSON();
 
    expect(renderedInputForm).toMatchSnapshot();
  });
});
