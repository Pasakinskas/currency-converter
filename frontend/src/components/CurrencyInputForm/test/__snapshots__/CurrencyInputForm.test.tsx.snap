// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencyInputForm renders correctly 1`] = `
<div
  className="pure-form pure-form-aligned"
>
  <div
    className="pure-control-group"
  >
    <label
      htmlFor="initial-amount"
    >
      Initial Amount
    </label>
    <input
      className="pure-u-1-5"
      id="initial-amount"
      min="0"
      onChange={[Function]}
      required={true}
      step="0.01"
      type="number"
    />
  </div>
  <div
    className="pure-control-group"
  >
    <label
      htmlFor="initial-currency"
    >
      Initial Currency
    </label>
    <select
      className="pure-u-1-5"
      defaultValue="DEFAULT"
      id="initial-currency"
      onChange={[Function]}
    >
      <option
        disabled={true}
        hidden={true}
        value="DEFAULT"
      >
        -
      </option>
    </select>
  </div>
  <div
    className="pure-control-group"
  >
    <label
      htmlFor="target-currency"
    >
      Target Currency
    </label>
    <select
      className="pure-u-1-5"
      defaultValue="DEFAULT"
      id="target-currency"
      onChange={[Function]}
    >
      <option
        disabled={true}
        hidden={true}
        value="DEFAULT"
      >
        -
      </option>
    </select>
  </div>
  <div
    className="pure-control-group"
  >
    <label
      htmlFor="conversion-rate"
    >
      Conversion Rate
    </label>
    <input
      className="pure-u-1-5"
      id="conversion-rates"
      readOnly={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="pure-control-group"
  >
    <label
      htmlFor="amount-after-conversion"
    >
      Amount after Conversion
    </label>
    <input
      className="pure-u-1-5"
      id="amount-after-conversion"
      readOnly={true}
      type="text"
    />
  </div>
  <div
    className="pure-controls"
  >
    <button
      className="pure-button pure-button-primary"
      onClick={[Function]}
    >
      Calculate
    </button>
  </div>
</div>
`;
